src/sonic-yang-models/yang-models/sonic-bgp-neighbor.yang

```diff
@@ -0,0 +1,108 @@
+module sonic-bgp-neighbor {
+    namespace "http://github.com/Azure/sonic-bgp-neighbor";
+    prefix sbnbr;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-bgp-common {
+        prefix sbc;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+    import sonic-bgp-global {
+        prefix sbgpg;
+    }
+
+    import sonic-bgp-peergroup {
+        prefix pg;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP Neighbor";
+
+    revision 2021-02-26 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-neighbor {
+        container BGP_NEIGHBOR {
+            list BGP_NEIGHBOR_LIST {
+                key "vrf_name neighbor";
+
+                leaf vrf_name {
+                    type leafref {
+                        path "/sbgpg:sonic-bgp-global/sbgpg:BGP_GLOBALS/sbgpg:BGP_GLOBALS_LIST/sbgpg:vrf_name";
+                    }
+                    description "Network-instance/VRF name";
+                }
+
+                leaf neighbor {
+                    type union {
+                        type inet:ip-address;
+                        type leafref {
```

Comment:
Please handle subinterface cases too. For example, BGP neighbor could be configured for "Ethernet16.400" and the Yang should be able to handle this

Reply:
We'll add it once the subinterfaces YANG is available, not as part of this initial PR.